auth/utils/index.ts

```diff
@@ -14,6 +14,23 @@
  * limitations under the License.
  */
 
+/**
+ * @module utils
+ * Helper functions used by generated SDKs.
+ *
+ * Identity and Access Management (IAM)
+ * Cloud Pak for Data
+ *
+ * The token managers sit inside of an authenticator and do the work to retrieve
+ * tokens where as the authenticators add these tokens to the actual request.
+ *
+ * functions:
+ *   getAuthenticatorFromEnvironment: Get authenticator from external sources.
+ *   readExternalSources: Get config object from external sources.
```

Comment:
While readExternalSources is used by the authenticator factory, it is also used to retrieve service-level config information.  It might make sense to put this in a new "config" module such that it is used by both the auth function and the BaseService.  Just thinking out loud here :)

Reply:
Heard you loud and clear 👍